packages/cdk8s-plus-17/src/pod.ts

```diff
@@ -112,29 +112,33 @@ export class PodSpec implements IPodSpec {
       throw new Error('PodSpec must have at least 1 container');
     }
 
-    const volumes: k8s.Volume[] = [];
+    const volumes: Map<string, k8s.Volume> = new Map();
     const containers: k8s.Container[] = [];
 
     for (const container of this.containers) {
 
       // automatically add volume from the container mount
       // to this pod so thats its available to the container.
       for (const mount of container.mounts) {
-        volumes.push(mount.volume._toKube());
+        if (!volumes.has(mount.volume.name)) {
+          volumes.set(mount.volume.name, mount.volume._toKube());
+        }
       }
 
       containers.push(container._toKube());
     }
 
     for (const volume of this._volumes) {
-      volumes.push(volume._toKube());
+      if (!volumes.has(volume.name)) {
+        volumes.set(volume.name, volume._toKube());
```

Comment:
Should we ensure this is actually the same volume (toKube returns the same thing or the Volume instance is the same)? If they don't match I'd fail

Reply:
Good idea